src/Http/BaseApplication.php

```diff
@@ -73,6 +120,8 @@ public function routes($routes)
         if (!Router::$initialized) {
             require $this->configDir . '/routes.php';
             // Prevent routes from being loaded again
+
+            $this->pluginRegistry()->routes($routes);
```

Comment:
I would prefer to see us load routes in the application's routes file. This allows the app developer the ability to add additional paths/parameters via scopes. In 3.5 we added `RouteBuilder::loadPlugin()` to load routes from a single plugin. If the route builder had access to the plugin registry, or application instance we'd be able to leverage that existing interface.  With that said we may also need a `$routes->loadPlugins()` to cover the case where people just don't want to customize routes and just load them all.

Reply:
Any suggestion how to have a single plugin registry and share it between the router and the app without having a global DI container?

Reply:
We could hide the PluginRegistry inside `Plugin`. That would let us make the new/old ways to work with plugins easier to deal with.

> I understand your arguments but it don't think this functionality should exist in two places?

I don't think we should plan to have the same functionality in two places as a permanent solution. Doing so requires us to document when to use each solution and ensure that both of them continue working.

> Also it's already loading the routes if the plugin is configured to load them? I don't see the issue except that we would lose a previously added methods feature?

The routebuilder methods make it easier to customize plugin routes with additional scopes/paths. For example

```php
$routes->scope('/tools', function ($routes) {
  $routes->loadPlugin('DebugKit');
});
```
Is not possible with the current implementation you have. This approach prevents plugin routes to be placed above routes defined in the application's route list. Placing frequently used routes higher in the file helps improve parse performance as fewer routes are checked on incoming requests.

Reply:
From the lack of a better designed solution making `Plugin` a “facade” could work for the time being. We can rethink that in the future (4.x/5.x) when we have a DIC built in.

Reply:
>  In 3.5 we added RouteBuilder::loadPlugin() to load routes from a single plugin. 

Can't we check in the method if the first passed was a plugin registry object and then use the 2nd args as string for the name?

> $routes->loadPlugins()

Could take the plugin registry as arg as well.